added return types to openmc.Settings

shimwell opened the pull request:
This PR aims to add return types to the ```openmc.Settings``` object.

This was originally proposed in PR #2016 along with moving the docstrings, however the docstring movement was not quite able to reproduce everything needed.

Now that #2016 is closed this aspect of #2016 has been split out so that is can be proposed separately  as discussed with @paulromano

paulromano commented on openmc/settings.py:
```diff
@@ -298,191 +298,191 @@ def __init__(self):
 
     @property
-    def rel_max_lost_particles(self):
+    def rel_max_lost_particles(self) -> int:
```
Looks like this one is wrong here and elsewhere as well. Can you also fix the docstring above and the type annotation in the setter?
```suggestion
    def rel_max_lost_particles(self) -> float:
```

paulromano commented on openmc/settings.py:
```diff
@@ -298,191 +298,191 @@ def __init__(self):
 
     @property
-    def source(self):
+    def source(self) -> Union[Source, typing.Iterable[Source]]:
```
Although the setter allows you to pass a scalar, it's always turned into a list
```suggestion
    def source(self) -> typing.List[Source]:
```

paulromano commented on openmc/settings.py:
```diff
@@ -298,191 +298,191 @@ def __init__(self):
 
     @property
-    def track(self):
+    def track(self) -> typing.Iterable[int]:
```
```suggestion
    def track(self) -> typing.Iterable[typing.Iterable[int]]:
```

paulromano commented on openmc/settings.py:
```diff
@@ -298,191 +298,191 @@ def __init__(self):
 
     @property
-    def volume_calculations(self):
+    def volume_calculations(self) -> Union[VolumeCalculation, typing.Iterable[VolumeCalculation]]:
```
```suggestion
    def volume_calculations(self) -> typing.List[VolumeCalculation]:
```

paulromano commented on openmc/settings.py:
```diff
@@ -298,191 +298,191 @@ def __init__(self):
 
     @property
-    def weight_windows(self):
+    def weight_windows(self) -> Union[WeightWindows, typing.Iterable[WeightWindows]]:
```
```suggestion
    def weight_windows(self) -> typing.List[WeightWindows]:
```

shimwell commented:
Thanks for the review @paulromano , I've added your suggestions and fixed the rel_max_lost_particles elsewhere in the code (docstring and setter type).
